File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.py
class Solution:
    def removeStars(self, s: str) -> str:
        length = len(s)
        stack = []
        res = []
        for i in range(len(s) - 1, -1, -1):
            if s[i] == '*':
                stack.append('*')
            else:
                if stack:
                    stack.pop()
                else:
                    res.append(s[i])
        return "".join(reversed(res))

File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.cpp
class Solution {
public:
    string removeStars(string s) {
        string store;
        for (auto c : s) {
            if (c != '*') {
                store.push_back(c);
            } else {
                if (!store.empty()) {
                    store.pop_back();
                }
            }
        }
        return store;
    }
};